src/Microsoft.AspNet.Server.Kestrel/Http/MemoryPoolBlock2.cs

```diff
@@ -0,0 +1,551 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Linq;
+using System.Numerics;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Microsoft.AspNet.Server.Kestrel.Http
+{
+    public class MemoryPoolBlock2
```

Comment:
This class needs some **UBER** unit testing. The enumerating over blocks should be easy to verify. All the edge cases :smile: